functions/src/index.ts

```diff
@@ -261,5 +262,45 @@ export const getStatistics = functions.https.onCall(async (data, context) => {
   throw new functions.https.HttpsError("invalid-argument", "invalid provider");
 });
 
+export const getUserEmotes = functions.https.onCall(async (data, context) => {
+  if (!context.auth) {
+    throw new functions.https.HttpsError("permission-denied", "missing auth");
+  }
+  const provider = data?.provider;
+  const channelId = data?.channelId;
+  if (!provider || !channelId) {
+    throw new functions.https.HttpsError(
+      "invalid-argument",
+      "missing provider, channelId"
+    );
+  }
+
+  switch (provider) {
+    case "twitch":
+      const channel = await getTwitchLogin(context.auth.uid, channelId);
+      if (!channel) {
+        return;
+      }
+      const client = await getTwitchClient(context.auth.uid, channelId);
+      await client.connect();
+
+      var retrieveEmotes = new Promise(function (resolve) {
+        client.on("emotesets", (set: string, emotes: EmoteObj) => resolve(emotes));
+      })
+
+      try {
+        const emoteInfo: any = await retrieveEmotes;
+        var emoteList: any = [];
+        Object.values(emoteInfo).forEach((value: any) => Array.prototype.push.apply(emoteList, value));
+        return { emotes: emoteList };
+      }
+      finally {
```

Comment:
Add a `catch` block to throw a different `HttpsError`, otherwise `invalid provider` will get thrown here.